tee-worker/litentry/core/credentials/src/nodereal/amount_holding/evm_amount_holding.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright 2020-2023 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+use litentry_primitives::{all_evm_web3networks, EVMTokenType, Web3Network};
+
+use crate::{
+	assertion_logic::{AssertionLogic, Op},
+	litentry_profile::{BalanceRange, BalanceRangeIndex},
+	Credential,
+};
+pub trait EVMTokenAddress {
+	fn get_address(&self, network: Web3Network) -> Option<&'static str>;
+}
+
+impl EVMTokenAddress for EVMTokenType {
+	fn get_address(&self, network: Web3Network) -> Option<&'static str> {
+		match (self, network) {
+			(EVMTokenType::Ton, Web3Network::Bsc) =>
+				Some("0x76a797a59ba2c17726896976b7b3747bfd1d220f"),
+			(EVMTokenType::Ton, Web3Network::Ethereum) =>
+				Some("0x582d872a1b094fc48f5de31d3b73f2d9be47def1"),
+			(EVMTokenType::Trx, Web3Network::Bsc) =>
+				Some("0xCE7de646e7208a4Ef112cb6ed5038FA6cC6b12e3"),
+			(EVMTokenType::Trx, Web3Network::Ethereum) =>
+				Some("0x50327c6c5a14dcade707abad2e27eb517df87ab5"),
+			_ => None,
+		}
+	}
+}
+
+const EVM_HOLDING_AMOUNT_RANGE: [f64; 8] =
+	[0.0, 100.0, 200.0, 500.0, 800.0, 1200.0, 1600.0, 3000.0];
+
+const TYPE: &str = "Token Holding Amount";
+const DESCRIPTION: &str = "The amount of a particular token you are holding";
+
+struct AssertionKeys {
+	token: &'static str,
```

Comment:
Any preference of `&'static str,` vs `String` ?

Reply:
These assertion keys are static and immutable, so using &'static str is more suitable than String.